### ISSUE TITLE
Units bombarded down to 0hp should die

### ISSUE THREAD

**Issue by hugomflavio** [longturn/freeciv21#2382]
**Describe the bug**
Extensively bombarded units inside a city. To the point they should have *0hp* left - especially since they have 0 defence. But the ground troops who then ended them reported winning without taking out any hp. see screenshot below.

**To Reproduce**
1. Let an idle city produce 10+ raiders so their defence becomes 0 (idk the actual endpoint that makes i t0)
2. bombard that city with 10 archers, unload them all.
3. attack the city with your own raiders after.

**Expected behavior**
defenders should have 1hp left?
edit: *they should die when bombarded down to 0*

**Screenshots**
![image](https://github.com/user-attachments/assets/93cf96ca-2ff3-448b-b090-27a7f6026d36)

**Platform and version (please complete the following information):**
 - OS: Linux
 - Freeciv21 version: 3.1-beta.2
 - Ruleset/Longturn game (if applicable): Sim06

**Additional context**
noticed during the test game

*Edit:*

* `Bombard_Limit_Pct` has a value of 0 in the ruleset (it is not set)
* See comment in #867

**Comment by hugomflavio** [comment]
(could be a bug in the message itself, of course)

**Comment by hugomflavio** [comment]
not related to the 0 defence. But it seems that somehow if one bombards units hard enough, they can accidentally get to 0hp? Need to test with eyes on both sides to understand what's going on.

**Comment by hugomflavio** [comment]
![image](https://github.com/user-attachments/assets/45755748-0393-4b5b-afbd-c138f6daa3f5)

Confirmed units are getting to 0hp from bombarding. Is this expected?

**Comment by lmoureaux** [comment]
No. I suppose `Bombard_Limit_Pct` is missing from the ruleset and `+Bombard_Limit_Pct` is set.

Bombarded units with 0hp should die.

**Comment by Corbeau-1** [comment]
I can confirm that this does happen, units do get to 0 HP while surviving. Technically, this is what bombardment does: bombs them to oblivion without physically destroying them. Now, we may discuss if this is the *desired* behaviour and, if not, what to do and how to do it. I'd be in favour of defining some minimal HP value that units can be bombed to (maybe make it an effect so that it depends on the unit, terrain, veterancy...) but I'll survive if the current condition remains.

**Comment by lmoureaux** [comment]
Edited the description to include the results of our investigation on Discord. Units bombarded down to 0hp should die, just like units reaching 0hp in combat.

**Comment by XHawk87** [comment]
I was thinking of looking into this soon. Presumably we'll want a message along with this to say what happened. Any suggestions for its content?

**Comment by lmoureaux** [comment]
Would start from:
```cpp
          /* TRANS: "Your green Warriors [id:100 ...D:1.0 HP:10]
           * helplessly perished in an attack by the Greek green Legion
           * [id:200 ...A:4.0 lost 1 HP, has 9 HP remaining%s]."
           * last %s is either ", promoted ..." or empty string */
          _("Your %s %s [id:%d %sD:%.1f HP:%d] helplessly perished in an"
            " attack by the %s %s %s [id:%d %s%sA:%.1f HP:%d%s]."),
```
Then change "attack" to something more fitting and adjust stats if it makes sense

**Comment by XHawk87** [comment]
I went with:
```
      /* TRANS: "Your green Warriors [id:100 D:1.0 HP:10]
       * helplessly perished in a bombardment by the Greek green Catapult
       * [id:200 ...A:4.0 FP:1 BR:10]." */
      _("Your %s %s [id:%d D:%.1f HP:%d] helplessly perished in a"
        " bombardment by the %s %s %s [id:%d A:%.1f FP:%d BR:%d]."),
```
Tested it and it's working. However, I now wonder if there should be a message for the attacker about units they see that they killed, to mirror the attack defender killed notification to the attacker.
